frontend/src/tests/components/Utils/DateConversion.test.js

```diff
@@ -1,42 +1,378 @@
 import { calculateTimezoneOffset, calculateSign, calculateHours, calculateMinutes, toLocalISOString, DateConversion } from "main/components/Utils/DateConversion";
 
+var assert = require('assert');
+
+var _Date = null;
+exports._Date = Date;
+
+var mockDateOptions = {};
+
+var timezone;
+
+var weekDays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+var months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'];
+
+var HOUR = 60 * 60 * 1000;
+
+var date_iso_8601_regex = /^\d\d\d\d(-\d\d(-\d\d(T\d\d:\d\d(:\d\d)?(\.\d\d\d)?(\d\d\d)?(Z|[+-]\d\d:?\d\d))?)?)?$/;
+var date_with_offset = /^\d\d\d\d-\d\d-\d\d( \d\d:\d\d:\d\d(\.\d\d\d)? )?(Z|(-|\+|)\d\d:\d\d)$/;
+var date_rfc_2822_regex = /^\d\d-\w\w\w-\d\d\d\d \d\d:\d\d:\d\d (\+|-)\d\d\d\d$/;
+var local_date_regex = /^\d\d\d\d-\d\d-\d\d[T ]\d\d:\d\d(:\d\d(\.\d\d\d)?)?$/;
+var local_GMT_regex = /^(Mon|Tue|Wed|Thu|Fri|Sat|Sun), \d\d \w\w\w \d\d\d\d \d\d:\d\d:\d\d GMT$/
+
+function MockDate(param) {
+  if (arguments.length === 0) {
+    this.d = new _Date();
+  } else if (arguments.length === 1) {
+    if (param instanceof MockDate) {
+      this.d = new _Date(param.d);
+    } else if (typeof param === 'string') {
+      if (param.match(date_iso_8601_regex) ||
+        param.match(date_with_offset) ||
```

Comment:
This seems like a lot of custom code to do something that can probably be done by a library that's already tested, and is maintained by someone else.

What is the real problem you are trying to solve?

Reply:
See also: https://ucsb-cs156-s24.slack.com/archives/C06RPBBPPLJ/p1716606773975469

Reply:
My code is fetching the environment’s timezone for testing. And the expected date/time is a fixed number when made the test. The thing is my local machine, GitHub action container are on two different timezone, which means the result it get will not match the number I put in the expected field. In my previous test, I try to match this number from the result I got from GitHub action test. But after that branch is being merged to main, and when we pulled from main to make our new branch, the local tests start to failing for date conversion. What I’m trying to do is fixing a timezone for the test file, and I did not know I could import a package by adding it into the package.json file. So I just manually added the tz database and required function for the mocking.

Reply:
I’ll modify relevant files to import the timezone-mock package to complete this test.